lib/parsers.js

```diff
@@ -287,15 +287,30 @@ exports.parseColor = function parseColor(val) {
     alpha = 1;
   var parts;
   var res = colorRegEx1.exec(val);
-  // is it #aaa or #ababab
+  // is it #aaa, #ababab, #aaaa, #abababaa
   if (res) {
+    var defaultHex = val.substr(1);
     var hex = val.substr(1);
-    if (hex.length === 3) {
+    if ([3, 4].indexOf(defaultHex.length) > -1) {
       hex = hex[0] + hex[0] + hex[1] + hex[1] + hex[2] + hex[2];
+
+      if (defaultHex.length === 4) {
```

Comment:
While it won't impact performance too negatively, having an additional copy of a string just to check its length is a bit confusing. The logic will be a bit more clear if we just save the length at the beginning:

```
var hex = val.substr(1);
var length = hex.length;
```

and then we can use that in the previous (3 and 4 comparison) and this (4 comparison) check.

Reply:
In fact I also use it to capture the default alpha and duplicate it to the end result because the 'hex' variable is already modified. What to do in this case?

`
if (defaultHex.length === 4) {
    hex = hex + defaultHex[3] + defaultHex[3];
}
`